AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString *const kMSUserDefaultsPrefix = @"MSAppCenter";
+
+/**
+ * Persistent settings, a wrapper around NSUserDefaults capable of updating object or dictionary (including expiration).
+ */
+@interface MSAppCenterUserDefaults : NSObject
+
+/**
+ * @return the shared settings.
+ */
++ (instancetype)shared;
+
+/**
+ * Returns an array of keys to be migrated.
+ */
++ (NSDictionary<NSString *, NSString *> *)keysToMigrate;
```

Comment:
It sounds like `keysToMigrate` and `migrateKeys` are only used by the `MSAppCenterUserDefaults` implementation and unit tests so should be in a private header by our convention.